Wordle.Api/Wordle.Api/Services/WordService.cs

```diff
@@ -1,65 +1,144 @@
-﻿using Microsoft.EntityFrameworkCore;
-using Wordle.Api.Data;
-
-namespace Wordle.Api.Services
-{
-    public class WordService
-    {
-        private readonly AppDbContext _db;
-
-        public WordService(AppDbContext db)
-        {
-            _db = db;
-        }
-
-        public async Task<string> GetRandomWord()
-        {
-            var count = await _db.Words.CountAsync(word => word.IsCommon);
-            var index = new Random().Next(count);
-            var word = await _db.Words
-                .Where(word => word.IsCommon)
-                .Skip(index)
-                .FirstAsync();
-            return word.Text;
-        }
-
-        public async Task<IEnumerable<Word>> GetSeveralWords(int? count)
-        {
-            count ??= 10;
-            var totalCount = await _db.Words.CountAsync(word => word.IsCommon);
-            totalCount -= count.Value;
-            int index = new Random().Next(totalCount);
-            var words = await _db.Words
-                .Where(word => word.IsCommon)
-                .Skip(index)
-                .Take(count.Value)
-                .OrderByDescending(w => w.Text)
-                .ToListAsync();
-            return words;
-        }
-
-        public async Task<Word> AddWord(string? newWord, bool isCommon)
-        {
-            if (newWord is null || newWord.Length != 5)
-            {
-                throw new ArgumentException("Word must be 5 characters long");
-            }
-            var word = await _db.Words.FirstOrDefaultAsync(w => w.Text == newWord);
-            if (word != null)
-            {
-                word.IsCommon = isCommon;
-            }
-            else
-            {
-                word = new()
-                {
-                    Text = newWord,
-                    IsCommon = isCommon
-                };
-                _db.Words.Add(word);
-            }
-            await _db.SaveChangesAsync();
-            return word;
-        }
-    }
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using Timer = System.Timers.Timer;
+using System.Timers;
+using Wordle.Api.Data;
+
+namespace Wordle.Api.Services;
+
+public class WordService
+{
+    private readonly AppDbContext _db;
+    private Word TodaysWord;
+    private readonly List<string> PreviousWords;
+
+    public WordService(AppDbContext db)
+    {
+        _db = db;
+        PreviousWords = new List<string>();
+        TodaysWord = new Word { Text = GetRandomWord().Result };
+        StartTimer();
+    }
+
+    public async Task<string> GetRandomWord()
+    {
+        if (TodaysWord.Text.IsNullOrEmpty())

```

Comment:
Getting an exception here when running locally. Several endpoints seem to cause the exception. (all from /word/*)